src/Components/Classes/Table/Table.js

```diff
@@ -0,0 +1,133 @@
+import React from 'react';
+import { useState } from 'react';
+
+const Table = ({ data, deleteItem, onEditItem, activities, trainers }) => {
+  const [klass, setKlass] = useState({
+    _id: '',
+    day: '',
+    hour: '',
+    activityId: '',
+    trainerId: '',
+    slots: ''
+  });
+
+  const onChangeInput = (e) => {
+    setKlass({
+      ...klass,
+      [e.target.name]: e.target.value
+    });
+  };
+
+  const funcionCargaDatos = (item) => {
+    setKlass({
+      _id: item._id,
+      day: item.day,
+      hour: item.hour,
+      activityId: item.activityId,
+      trainerId: item.trainerId,
+      slots: item.slots
+    });
+  };
+
+  const onSubmit = (e) => {
+    e.preventDefault();
+    onEditItem(klass._id, klass);
+    setKlass({
+      day: '',
+      hour: '',
+      activityId: '',
+      trainerId: '',
+      slots: ''
+    });
+  };
+
+  return (
+    <div>
+      <table>
+        <thead>
+          <tr>
+            <th>Activity Name</th>
+            <th>day</th>
+            <th>hour</th>
+            <th></th>
+            <th></th>
+          </tr>
+        </thead>
+        <tbody>
+          {data.map((item) => {
+            return (
+              <tr key={item._id}>
+                <td>{item.activityId.name}</td>
+                <td>{item.day}</td>
+                <td>{item.hour}</td>
+                <td>{item.slots}</td>
+                <td></td>
+                <td>
+                  <button className="editButton" onClick={() => funcionCargaDatos(item)}>
+                    Edit Item
+                  </button>
+                </td>
+                <td>
+                  <button
+                    className="deleteButton"
+                    onClick={() => {
+                      if (confirm('Are you sure that you want to delete this class?')) {
```

Comment:
You can declare that confirm as the value of a variable

Reply:
Do you mean something like this? `onClick={(confirm("confirm message") => { if(confirm) {...}`

Reply:
Try to declare a const inside that onClick and set its value to that confirm({message: 'your message'}). You can then do the if statement on that constant variable and do your alert logic inside it

Reply:
I have adjusted it like this. Is this what you meant?


![Captura](https://github.com/BaSP-m2023/ferpa-megarocket-app/assets/32039088/3ad7537f-1e16-4a17-84c0-add4228fbff8)